Error if running "make dist" without first creating a package directory

```
env DOCKER_IMAGE_TAG=centos-6-1.5.0 make dist
...
bash: -c: line 0: unexpected argument `]]' to conditional unary operator
bash: -c: line 0: syntax error near `;'
bash: -c: line 0: `if [[ -s /centos-ssh-apache-php.centos-6-1.5.0.tar.xz ]]; then           echo "--->  Saving package";        echo " --->  Package path: /centos-ssh-apache-php.centos-6-1.5.0.tar.xz";       echo " --->  Package already exists";   else        if [[ ! -d  ]]; then            echo "--->  Creating package directory";                mkdir -p ;          fi; echo "--->  Saving package";        /usr/local/bin/docker save              jdeathe/centos-ssh-apache-php:centos-6-1.5.0 |          /usr/local/bin/xz -9 >              /centos-ssh-apache-php.centos-6-1.5.0.tar.xz;               if [[ ${?} -eq 0 ]]; then                   echo " --->  Package path: /centos-ssh-apache-php.centos-6-1.5.0.tar.xz";           echo " --->  Package saved";                else                echo " --->  Package save error";                   exit 1;                 fi;     fi'
make: *** [dist] Error 2
```

--- a/.gitignore
+++ b/.gitignore
@@ -1,0 +1,1 @@
+packages

--- a/Makefile
+++ b/Makefile
@@ -18,9 +18,6 @@ PREFIX_SUB_STEP_POSITIVE := $(shell printf -- '%b%s%b' "$(COLOUR_POSITIVE)" "$(P
 
 .DEFAULT_GOAL := build
 
-# Get absolute file paths
-PACKAGE_PATH := $(realpath $(PACKAGE_PATH))
-
 # Package prerequisites
 docker := $(shell type -p docker)
 xz := $(shell type -p xz)
@@ -54,6 +51,7 @@ get-docker-info := $(shell $(docker) info)
 	require-docker-container-status-running \
 	require-docker-image-tag \
 	require-docker-release-tag \
+	require-package-path \
 	restart \
 	rm \
 	rmi \
@@ -99,23 +97,22 @@ create: prerequisites require-docker-container-not
 			exit 1; \
 		fi
 
-dist: prerequisites require-docker-release-tag | pull
-	@ if [[ -s $(PACKAGE_PATH)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz ]]; then \
+dist: prerequisites require-docker-release-tag require-package-path | pull
+	$(eval $@_package_path := $(realpath \
+		$(PACKAGE_PATH) \
+	))
+	@ if [[ -s $($@_package_path)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz ]]; then \
 			echo "$(PREFIX_STEP) Saving package"; \
-			echo "$(PREFIX_SUB_STEP) Package path: $(PACKAGE_PATH)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz"; \
+			echo "$(PREFIX_SUB_STEP) Package path: $($@_package_path)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz"; \
 			echo "$(PREFIX_SUB_STEP_POSITIVE) Package already exists"; \
 		else \
-			if [[ ! -d $(PACKAGE_PATH) ]]; then \
-				echo "$(PREFIX_STEP) Creating package directory"; \
-				mkdir -p $(PACKAGE_PATH); \
-			fi; \
 			echo "$(PREFIX_STEP) Saving package"; \
 			$(docker) save \
 				$(DOCKER_USER)/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG) | \
 				$(xz) -9 > \
-					$(PACKAGE_PATH)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz; \
+					$($@_package_path)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz; \
 				if [[ $${?} -eq 0 ]]; then \
-					echo "$(PREFIX_SUB_STEP) Package path: $(PACKAGE_PATH)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz"; \
+					echo "$(PREFIX_SUB_STEP) Package path: $($@_package_path)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz"; \
 					echo "$(PREFIX_SUB_STEP_POSITIVE) Package saved"; \
 				else \
 					echo "$(PREFIX_SUB_STEP_NEGATIVE) Package save error"; \
@@ -123,14 +120,17 @@ dist: prerequisites require-docker-release-tag | pull
 				fi; \
 		fi
 
-distclean: prerequisites require-docker-release-tag | clean
-	@ if [[ -e $(PACKAGE_PATH)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz ]]; then \
+distclean: prerequisites require-docker-release-tag require-package-path | clean
+	$(eval $@_package_path := $(realpath \
+		$(PACKAGE_PATH) \
+	))
+	@ if [[ -e $($@_package_path)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz ]]; then \
 			echo "$(PREFIX_STEP) Deleting package"; \
-			echo "$(PREFIX_SUB_STEP) Package path: $(PACKAGE_PATH)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz"; \
-			find $(PACKAGE_PATH) \
+			echo "$(PREFIX_SUB_STEP) Package path: $($@_package_path)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz"; \
+			find $($@_package_path) \
 				-name $(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz \
 				-delete; \
-			if [[ ! -e $(PACKAGE_PATH)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz ]]; then \
+			if [[ ! -e $($@_package_path)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz ]]; then \
 				echo "$(PREFIX_SUB_STEP_POSITIVE) Package cleanup complete"; \
 			else \
 				echo "$(PREFIX_SUB_STEP_NEGATIVE) Package cleanup failed"; \
@@ -157,15 +157,18 @@ logs-delayed: prerequisites
 	@ sleep 2
 	@ $(MAKE) logs
 
-load: prerequisites require-docker-release-tag
+load: prerequisites require-docker-release-tag require-package-path
+	$(eval $@_package_path := $(realpath \
+		$(PACKAGE_PATH) \
+	))
 	@ echo "$(PREFIX_STEP) Loading image from package"; \
-		echo "$(PREFIX_SUB_STEP) Package path: $(PACKAGE_PATH)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz"; \
-		if [[ ! -s $(PACKAGE_PATH)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz ]]; then \
+		echo "$(PREFIX_SUB_STEP) Package path: $($@_package_path)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz"; \
+		if [[ ! -s $($@_package_path)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz ]]; then \
 			echo "$(PREFIX_STEP_NEGATIVE) Package not found"; \
 			echo "$(PREFIX_SUB_STEP_NEGATIVE) To create a package try: DOCKER_IMAGE_TAG=\"$(DOCKER_IMAGE_TAG)\" make dist"; \
 			exit 1; \
 		else \
-			$(xz) -dc $(PACKAGE_PATH)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz | \
+			$(xz) -dc $($@_package_path)/$(DOCKER_IMAGE_NAME).$(DOCKER_IMAGE_TAG).tar.xz | \
 				$(docker) load; \
 			echo "$(PREFIX_SUB_STEP) $$( if [[ -n $$($(docker) images -q $(DOCKER_USER)/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)) ]]; then echo $$($(docker) images -q $(DOCKER_USER)/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)); else echo $$($(docker) images -q docker.io/$(DOCKER_USER)/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)); fi; )"; \
 			echo "$(PREFIX_SUB_STEP_POSITIVE) Image loaded"; \
@@ -263,6 +266,19 @@ require-docker-release-tag:
 	@ if [[ -z $$(if [[ $(DOCKER_IMAGE_TAG) =~ $(DOCKER_IMAGE_RELEASE_TAG_PATTERN) ]]; then echo $(DOCKER_IMAGE_TAG); else echo ''; fi) ]]; then \
 			echo "$(PREFIX_STEP_NEGATIVE) Invalid DOCKER_IMAGE_TAG value: $(DOCKER_IMAGE_TAG)"; \
 			echo "$(PREFIX_SUB_STEP) A release tag is required for this operation."; \
+			exit 1; \
+		fi
+
+require-package-path:
+	@ if [[ -n $(PACKAGE_PATH) ]] && [[ ! -d $(PACKAGE_PATH) ]]; then \
+			echo "$(PREFIX_STEP) Creating package directory"; \
+			mkdir -p $(PACKAGE_PATH); \
+		fi; \
+		if [[ ! $${?} -eq 0 ]]; then \
+			echo "$(PREFIX_STEP_NEGATIVE) Failed to make package path: $(PACKAGE_PATH)"; \
+			exit 1; \
+		elif [[ -z $(PACKAGE_PATH) ]]; then \
+			echo "$(PREFIX_STEP_NEGATIVE) Undefined PACKAGE_PATH"; \
 			exit 1; \
 		fi
 

